src/EventRegistry.h

```diff
@@ -106,6 +160,9 @@ class EventRegistry final {
 
     // Map event groups identified by kind and name to their instances.
     std::map<std::pair<EventGroupKind, std::string>, std::shared_ptr<EventGroup>, std::less<>> event_groups;
+
+    // Map for event metadata identifier to their descriptors types.
+    std::map<zeek_uint_t, EventMetadataDescriptor> event_metadata_types;
```

Comment:
Like mentioned above, this could probably be a set (I'd go with `unordered_set` since we do not care about exact order), but it should be faster for our use case (low probability of hash collisions). If you keep this a map let's make it an `unordered_map` with the same efficiency argument.